File: Binary Search/Binary.java
public class Binary {

    public static int search(int[] array, int number) {

        int low = 0;
        int high = array.length - 1;
        String guessToX = "";

        while (low <= high) {
            int mid = (high + low) / 2;
            int guess = array[mid];

            if (guess == number) {
                return mid;
            } else if (guess > number) {
                high = mid - 1;
                guessToX = "Too high";
            } else {
                low = mid + 1;
                guessToX = "Too low";
            }
            System.out.println("Current guess is " + guess + " and is " + guessToX);
        }
        return -1; // Return -1 if the number is not found in the array
    }

    public static void main(String[] args) {
        int[] myList = new int[10100];
        for (int i = 0; i <= 10100; i++) {
            myList[i] = i;
        }

        int result = search(myList, 44);
        if (result != -1) {
            System.out.println("Number found at index: " + result);
        } else {
            System.out.println("Number not found in the array.");
        }
    }
}

File: Binary Search/binary.js

const search = (array, number) => {
    
    let low = 0;
    let high = array.length - 1;
    let guessToX = "";

    while (low <= high) {
        let mid = Math.floor((high + low) / 2); 
        let guess = array[mid];

        if (guess === number) {
            return mid;
        } else if (guess > number) {
            high = mid - 1;
            guessToX = "Too high";
        } else {
            low = mid + 1;
            guessToX = "Too low";
        }
        console.log(`Current guess is ${guess} and is ${guessToX}`);
    }
    return null;
}

let myList = Array.from({ length: 10100 }, (_, index) => index);

console.log(search(myList, 44));
